src/XAMLTools.Core/XAMLCombine/XAMLCombiner.cs

```diff
@@ -188,19 +190,21 @@ public string Combine(IEnumerable<string> sourceFiles, string targetFile, bool i
                             }
 
                             var currentMergedSources = mergedDictionariesListNode.ChildNodes.OfType<XmlElement>()
-                                .Select(node => node.GetAttribute("Source"))
-                                .Where(source => !string.IsNullOrEmpty(source)).ToList();
+                                .Select(nodeElement => nodeElement.GetAttribute("Source"))
+                                .Where(source => !string.IsNullOrEmpty(source))
+                                .ToList();
 
                             foreach (var mergedDictionaryReference in xmlElement.ChildNodes)
                             {
-                                if (mergedDictionaryReference is not XmlElement mergedDictionaryReferenceElement || mergedDictionaryReferenceElement.Name != "ResourceDictionary")
+                                if (mergedDictionaryReference is not XmlElement mergedDictionaryReferenceElement || mergedDictionaryReferenceElement.Name != ResourceDictionaryString)
                                 {
                                     continue;
                                 }
 
                                 var sourceValue = mergedDictionaryReferenceElement.GetAttribute("Source");
                                 if (string.IsNullOrEmpty(sourceValue))
                                 {
+                                    Trace.WriteLine(string.Format($"Ignore merged ResourceDictionary inside resource \"{resourceFile}\""));
```

Comment:
Trace or BuildEngine....?

Reply:
BuildEngine is only available inside msbuild task, which we don't pass to XamlCombiner class